java/code/code.iml

```diff
@@ -0,0 +1,182 @@
+          <root url="jar://$USER_HOME$/.ivy2/cache/javax.xml.bind/jaxb-api/jars/jaxb-api-2.3.0.jar!/" />
+          <root url="jar://$USER_HOME$/.ivy2/cache/com.sun.xml.bind/jaxb-core/jars/jaxb-core-2.3.0.jar!/" />
+          <root url="jar://$USER_HOME$/.ivy2/cache/com.sun.xml.bind/jaxb-impl/jars/jaxb-impl-2.3.0.jar!/" />
+          <root url="jar://$USER_HOME$/.ivy2/cache/suse/junit/jars/junit-4.12.jar!/" />
+          <root url="jar://$USER_HOME$/.ivy2/cache/org.jmock/jmock/jars/jmock-2.12.0.jar!/" />
+          <root url="jar://$USER_HOME$/.ivy2/cache/org.jmock/jmock-junit3/jars/jmock-junit3-2.12.0.jar!/" />
+          <root url="jar://$USER_HOME$/.ivy2/cache/org.jmock/jmock-legacy/jars/jmock-legacy-2.12.0.jar!/" />
+          <root url="jar://$USER_HOME$/.ivy2/cache/org.jmock/jmock-imposters/jars/jmock-imposters-2.12.0.jar!/" />
+          <root url="jar://$USER_HOME$/.ivy2/cache/mockobjects/mockobjects-core/jars/mockobjects-core-0.09.jar!/" />
+          <root url="jar://$USER_HOME$/.ivy2/cache/mockobjects/mockobjects-jdk1.4-j2ee1.3/jars/mockobjects-jdk1.4-j2ee1.3-0.09.jar!/" />
+          <root url="jar://$USER_HOME$/.ivy2/cache/org.objenesis/objenesis/jars/objenesis-1.0.jar!/" />
+          <root url="jar://$USER_HOME$/.ivy2/cache/javax.websocket/javax.websocket-api/bundles/javax.websocket-api-1.1.jar!/" />
+          <root url="jar://$USER_HOME$/.ivy2/cache/strutstestcase/strutstestcase/jars/strutstestcase-2.1.3-1.2-2.4.jar!/" />
+          <root url="jar://$USER_HOME$/.ivy2/cache/org.jacoco/org.jacoco.ant/jars/nodeps-0.8.7.jar!/" />
+          <root url="jar://$USER_HOME$/.ivy2/cache/suse/hibernate-types/jars/hibernate-types-2.12.1.jar!/" />
+          <root url="jar://$USER_HOME$/.ivy2/cache/suse/jackson-annotations/jars/jackson-annotations-2.10.2.jar!/" />
+          <root url="jar://$USER_HOME$/.ivy2/cache/suse/jackson-core/jars/jackson-core-2.10.2.jar!/" />
+          <root url="jar://$USER_HOME$/.ivy2/cache/suse/jackson-databind/jars/jackson-databind-2.10.5.1.jar!/" />
+          <root url="jar://$USER_HOME$/.ivy2/cache/suse/salt-netapi-client/jars/salt-netapi-client-0.19.0.jar!/" />
+          <root url="jar://$USER_HOME$/.ivy2/cache/suse/snakeyaml/jars/snakeyaml-1.28.jar!/" />
+          <root url="jar://$USER_HOME$/.ivy2/cache/suse/pgjdbc-ng/jars/pgjdbc-ng-0.8.7.jar!/" />
+          <root url="jar://$USER_HOME$/.ivy2/cache/suse/spy/jars/spy-0.8.7.jar!/" />
+          <root url="jar://$USER_HOME$/.ivy2/cache/suse/client/jars/client-1.0.0~beta.2.jar!/" />
+          <root url="jar://$USER_HOME$/.ivy2/cache/suse/common/jars/common-1.0.0~beta.2.jar!/" />
+          <root url="jar://$USER_HOME$/.ivy2/cache/suse/hibernate-c3p0/jars/hibernate-c3p0-5.3.25.jar!/" />
+          <root url="jar://$USER_HOME$/.ivy2/cache/suse/hibernate-core/jars/hibernate-core-5.3.25.jar!/" />
+          <root url="jar://$USER_HOME$/.ivy2/cache/suse/hibernate-ehcache/jars/hibernate-ehcache-5.3.25.jar!/" />
```

Comment:
About this whole section I am not so sure: what if the versions are changing or differ locally, wouldn't this become a constant source of possible merge conflicts or at least unnecessary changes in patches? I guess that the ivy configuration file should probably be the only place where we define those version numbers checked into git.

Reply:
that is right... The issue is that some parts of that file are still needed to define the source tree (called `module` in IntelliJ's langage). I'll need to looks if something can be done for that